build/deploy/core-service.libsonnet

```diff
@@ -100,6 +100,7 @@ local ingress(metadata) = base.Ingress(metadata, 'https-ingress') {
                 },
               },
             },
+            imagePullSecrets: if metadata.image_pull_secret != null then [{name: metadata.image_pull_secret}] else [],
```

Comment:
I might be a bit rusty with tanka, but shouldn't you put this line in the definitions in `base.libjsonnet`? The `base.libjsonnet` definitions also have access to the metadata. This way that configuration option would apply to all deployments in the project. Also what about other manifests that might need this setting, like `StatefulSet`s?

Reply:
Good point, the initial intention was to limit the use of credentials to the image built in this repository but it seems that it can be passed to all resources pulling images even if it is using a different registry. I will update it.